src/eradiate/scenes/atmosphere/_particle_layer.py

```diff
@@ -182,8 +182,11 @@ def eval_width(self, ctx: KernelDictContext) -> pint.Quantity:
 
         else:
             if self.width is AUTO:
-                spectral_ctx = ctx.spectral_ctx if ctx is not None else None
-                return 10.0 / self.eval_sigma_s(spectral_ctx=spectral_ctx).min()
+                min_sigma_s = self.eval_sigma_s(spectral_ctx=ctx.spectral_ctx).min()
+                width = np.divide(
+                    1.0, min_sigma_s, where=min_sigma_s != 0.0, out=np.array([np.inf])
+                )
```

Comment:
Since this is not a properly vectorised call, why not use a more ordinary syntax? Like
```python
width = 1.0 / min_sigma_s if min_sigma_s != 0.0 else np.inf
```

Reply:
Of course, thanks!